app/src/main/java/org/mozilla/fenix/library/history/HistorySearchDialogFragment.kt

```diff
@@ -254,4 +264,49 @@ class HistorySearchDialogFragment : AppCompatDialogFragment(), UserInteractionHa
             }
         }
     }
+
+    private val startForResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+        result: ActivityResult ->
+        if (result.resultCode == Activity.RESULT_OK) {
+            val intent = result.data
+            intent?.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)?.first()?.also {
+                toolbarView.view.edit.updateUrl(url = it, shouldHighlight = true)
+                interactor.onTextChanged(it)
+                toolbarView.view.edit.focus()
+            }
+        }
+    }
+
+    private fun addVoiceSearchButton() {
+        if (voiceSearchButtonAlreadyAdded) return
```

Comment:
Let's overload this to escape early.

```suggestion
        if (voiceSearchButtonAlreadyAdded || !isVisible) return
```

Reply:
Sure, I'll update.  Thanks,